### PR TITLE
Fix #279 #280

### LINKED ISSUE #279: custom_db_roles cannot be created with only inherited roles
In the Atlas UI it is easily possible to create custom_db_roles which are comprised of only inherited roles. However, the resource definition currently requires one to always pass `actions`. Therefore, I don't see a chance to create roles which only use inherited roles such as this example:
image

### PR DESCRIPTION
**Pull request by PacoDw**
## Description

The inherited Roles attribute was changed to Set type to handle the best way when a user wants to remove/update inherited roles.

Link to any related issue(s): #279 #280 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by themantissa — CHANGES_REQUESTED** [review]
Looks good except failed checks need to be fixed so not yet approving.

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go [hunk 1]
```diff
@@ -331,23 +332,21 @@ func flattenActionResources(resources []matlas.Resource) []map[string]interface{
 }
 
 func expandInheritedRoles(d *schema.ResourceData) []matlas.InheritedRole {
-	var inheritedRoles []matlas.InheritedRole
+	vIR := d.Get("inherited_roles").(*schema.Set).List()
```
Comment (languitar) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go:334]: I would propose to use speaking variable names.

File: mongodbatlas/resource_mongodbatlas_custom_db_role_test.go [hunk 2]
```diff
@@ -70,7 +69,7 @@ func TestAccResourceMongoDBAtlasCustomDBRoles_WithInheritedRoles(t *testing.T) {
 
 	inheritRole := []matlas.CustomDBRole{
 		{
-			RoleName: fmt.Sprintf("a_test-acc-INHERITED_ROLE-%s", acctest.RandString(5)),
+			RoleName: fmt.Sprintf("test-acc-INHERITED_ROLE-%s", acctest.RandString(5)),
```
Comment (languitar) [inline mongodbatlas/resource_mongodbatlas_custom_db_role_test.go:72]: Why is this change necessary?

File: mongodbatlas/resource_mongodbatlas_custom_db_role_test.go [hunk 3]
```diff
@@ -556,8 +731,6 @@ func testAccMongoDBAtlasCustomDBRolesConfigBasic(projectID, roleName, action, da
 }
 
 func testAccMongoDBAtlasCustomDBRolesConfigWithInheritedRoles(projectID string, inheritedRole []matlas.CustomDBRole, testRole *matlas.CustomDBRole) string {
-	log.Printf("LOG___ inheritedRole: %#+v\n", inheritedRole[1])
```
Comment (languitar) [inline mongodbatlas/resource_mongodbatlas_custom_db_role_test.go:559]: Why this change?

File: mongodbatlas/resource_mongodbatlas_custom_db_role_test.go (hunk 2 above)
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_custom_db_role_test.go:72]: well reviewing the acceptance test I checked this, I think is better write the test correctly instead of this

File: mongodbatlas/resource_mongodbatlas_custom_db_role_test.go (hunk 3 above)
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_custom_db_role_test.go:559]: Test logs shouldn't be present in the code, it's not necessary

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go (hunk 1 above)
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_custom_db_role.go:334]: I can change this more clarify but in golang have this kind of variables is very common, it's a convention write short variable names.

**Review by themantissa — APPROVED** [review]
LGTM - hopefully user is able to test it fixes the reported issues - needs DoU reviewer before merge.

**Review by coderGo93 — APPROVED** [review]
LGTM
